### ISSUE TITLE
RichTextInput doesn't update on changes caused by redux-form actions

### ISSUE THREAD

**Issue by manolisan** [marmelab/react-admin#2464]
**What you were expecting:**
In one view with two **RichTextInput** components with the **same source**, changing the first one should change the value of the other and vise versa. 
However it is working as expected with two **TextInput** components with the **same source**, and they are in sync.

**What happened instead:**
Both RichTextInput component populate data on the same record in redux, but each component renders only what is typed within in it and not the value of the redux form.

**Steps to reproduce:**
Simply put two RichTextInput components with the same source in one SimpleForm.

**Other information:**

- Looking at the code, it appears that the Quill instance is populated with data only on mount. So, when new data comes asynchronously from REDUX, the RichTextInput keeps it's initial value.
 - Remounting a RichTextInput component leads to rendering the value of the redux form.

**Environment**

* React-admin version: 2.3.3
* React version: 16.3.2
* Browser: Chrome 69.0

**Comment by afilp** [comment]
@fzaninotto I believe that this closed (not merged) PR solves this problem, can you please re-consider this? Thank you!

**Comment by fzaninotto** [comment]
The patch in #2581 didn't seem to solve the problem. If you can provide a unit test or a CodeSandbox proving that it works, I'm willing to reconsider merging it.

**Comment by manolisan** [comment]
You can check the behavior while creating a new post in this [CodeSandbox](https://codesandbox.io/s/k0328o1v13).

The problem appears on the RichTextInput for engish. We have 2 of them, one in the typical create page and another one (with exaclty the same source) inside the dialog that appears when you click the translations button. 

The same example with TextInput instead of RichTextInput works seamless and the 2 inputs are synchronized.
